internal/xds/translator/oidc.go

```diff
@@ -210,9 +220,55 @@ func oauth2Config(oidc *ir.OIDC) (*oauth2v3.OAuth2, error) {
 		oauth2.Config.Credentials.CookieDomain = *oidc.CookieDomain
 	}
 
+	// Set the retry policy if it exists.
+	if oidc.Provider.Traffic != nil && oidc.Provider.Traffic.Retry != nil {
+		var rp *corev3.RetryPolicy
+		if rp, err = buildNonRouteRetryPolicy(oidc.Provider.Traffic.Retry); err != nil {
+			return nil, err
+		}
+		oauth2.Config.RetryPolicy = rp
+	}
 	return oauth2, nil
 }
 
+func buildNonRouteRetryPolicy(rr *ir.Retry) (*corev3.RetryPolicy, error) {
```

Comment:
can't we reuse the existing helper func and just ignore/unset `PerRetry.Timeout` & `RetryOn.HTTPStatusCodes` ?

Reply:
They use different Retry types: `routev3.RetryPolicy` vs `corev3.RetryPolicy`.